### ISSUE TITLE
False positive [E0415] "parameter bound more than once" for trait function

### ISSUE THREAD

**Issue by panstromek** [intellij-rust/intellij-rust#6191]
This code is valid but it produces an error "Identifier `x` is bound more than once in this parameter list".

```rust
pub trait Foo {
    fn bar(x: i32, x: i32);
}
```
 Parameter can be bound more than once in trait definitions. Kinda weird, though. I wouldn't expect this to be valid myself.
